src/core/src/index/revindex/disk_revindex.rs

```diff
@@ -351,7 +351,15 @@ impl RevIndexOps for RevIndex {
                         .collection
                         .record_for_dataset(dataset_id)
                         .expect("dataset not found");
-                    Some((row.name().into(), size))
+
+                    let mut name = row.name();
```

Comment:
This is ugly - is there an idiomatic way to do this??

Reply:
yuck: https://stackoverflow.com/questions/45343320/how-can-i-return-the-first-non-empty-string

Reply:
maybe?

```
                    let name = [row.name(), row.filename(), row.md5()]                                                                                                                               
                        .into_iter()                                                                                                                                                                 
                        .skip_while(|v| v.is_empty())                                                                                                                                                
                        .next()                                                                                                                                                                      
                        .unwrap();  // guaranteed to succeed because `md5` always exists
```

Reply:
yeh.... :)

Reply:
in 0ed80c1a